projects/laji/src/app/shared-modules/observation-map/observation-map/observation-map.component.ts

```diff
@@ -85,6 +91,30 @@ const BOX_QUERY_AGGREGATE_LEVELS = [
   ['gathering.conversions.wgs84Grid005.lat', 'gathering.conversions.wgs84Grid01.lon']
 ];
 
+
+const activeColor = '#6ca31d';
+
+const getFeatureStyleWithHoverAndActive = (getFeatureStyle: DataOptions['getFeatureStyle']) => (params: GetFeatureStyleOptions) => {
+  const {active, hovered} = params;
+  const style = getFeatureStyle(params);
+  const isActiveBox =params.feature.geometry.type !== 'Point' && active;
+  // Active point is styled with classname, boxes receive color in the style object.
+  // This is because the Leaflet Path objects don't support updating the class name.
+  const baseColor = isActiveBox
+    ? activeColor
+    : style.color;
+  const maxColorChangeDecimal = 100;
+  const color = combineColors(baseColor, ...(hovered ? ['#fff'] : []), maxColorChangeDecimal); // Highlight hovered item.
+  const className = [style.className, active && !isActiveBox ? 'laji-map-active-pointer' : undefined].filter(s => s).join(' ');
+  const _style = {...style, color, className};
+  if (isActiveBox) {
+    _style.weight = 3;
+    _style.color = color;
+    _style.fillColor = combineColors(style.color, activeColor, maxColorChangeDecimal); // Slide color towards active color.
+  }
+  return _style;
```

Comment:
I think currying is confusing in general. This could probably work better as a function decorator or something, but it's probably fine as it is.

Reply:
I'd probably call it augmentGetFeatureStyleWithHoverAndActive or decorateGetFeatureStyleWithHoverAndActive  or something like that since otherwise it's easy to confuse with the actual getFeatureStyle

Reply:
A function decorator is just a curry function with syntactic sugar, and it couldn't be used here with the `@` syntax since it is done dynamically. I renamed it anyway, good idea.

Reply:
Couldn't it? If say `getFeatureStyle` was a class method, then you annotate it with a method decorator `augmentHoverAndActive`. Then when passing getFeatureStyle to dataoptions you'd `.bind(this)`. I thought this would work, but I haven't tested it.

Reply:
Well yes it could be done, but that would change the logic quite a bit as you then must have a single class method `getFeatureStyle`. I started trying that out, and in the end neither a curry fn or a decorator was needed. It escalated a lot... :D Here's a list of what changed:

* Only a single data options getter is used, both points & aggregate boxes use it.
* Points & aggregate box getters return just the feature collections. The box cache also caches just the feature collections, as the data options were anyway modified after a cache hit.  I think it's kind of more clean now that the data flows as feature collections, and the transformation into laji-map consumable data is done as late as possible.
* The feature click handler is done now by a proxy method that directs the event to the correct handler based on the feature type. Not necessarily an improvement, but a consequence of the other changes.

What do you think? I'm not completely sure if it seems better to me just because I wrote it and thus it's more comprehensible for me.

...The topic of whether to use a class method decorator vs a curry function  is a choice of personal preference to some extent. Both have use cases where the other might suite better, but in general I prefer simple currying for several reasons:
* more agnostic since it can be used for all functions, not just class methods
* for me they are much more easy to write - for a method decorator I'd have to look up the signature from the docs and even then it's just more complex (as a side note I think class decorators don't suffer from this though)
* it's plain js, where as decorators are an experimental feature. I've actually ran into problems with them when I've changed the TS compiler on laji-map (I think the execution order of the method decorators was different between compilers). If used only inside an Angular project, this isn't a problem of course.

Reply:
Looks good! I think this is definitely the right approach.

About curry vs decorator: The thing I like about decorators is that the behavior is explicit. When I see a decorator I know immediately that it takes a function, returns a function etc. All of your reasons are totally fair and true though.